src/Libraries/Inscriptions/Envelope.ts

```diff
@@ -60,16 +61,26 @@ export class Envelope {
    * @param tx - Raw bitcoin transaction to search for an inscription envelope
    */
   static fromTransaction(tx: RawTransaction) {
-    const [data, oip] = getEnvelopeDataFromTx(tx) ?? [];
-    if (data) {
-      return new Envelope(tx.txid, data, oip);
+    const envelopes = getEnvelopesDataFromTx(tx);
+    if (envelopes) {
+      return envelopes.map(([data, oip], index) => {
+        if (data) {
+          return new Envelope(tx.txid, data, oip, index);
+        }
+        return null;
```

Comment:
Do we need to add null on the array? Or can we just skip it

Reply:
skip it,  the inscription indexer will ignore it if the array undefined/null